File: 2022-2023-2/Compilation-Principle-and-Design/Lab-1/Java/src/algorithm/quicksort.java
package src.algorithm;

import src.algorithm.partition;

public class quicksort {
	
	public int[] quicksort(int[] arr, int low, int high) {
		
		int mid;
		
		if (low < high) {
			
			partition _partition = new partition(arr, low, high);
			
			arr = _partition.getArray();
			mid = _partition.getX();
			
			quicksort(arr, low    , mid - 1);
			quicksort(arr, mid + 1, high   );
		}
		
		return arr;
	}
}

File: 2022-2023-2/Compilation-Principle-and-Design/Lab-1/Python/src/algorithm/quicksort.py
from . import partition

def quicksort(arr, low, high):
    
    if low < high:
        
        arr, mid = partition.partition(arr, low, high)
        
        quicksort(arr, low    , mid - 1)
        quicksort(arr, mid + 1, high   )
        
    return arr